### PR TITLE
Add Object_Id support for Managed Identity

### PR DESCRIPTION
**Pull request by gladjohn**
Partially Fixes (code fix) #4156 - docs and sample updates needs to be done.

Add Object_Id support for Managed Identity

### CODE REVIEW

File: tests/Microsoft.Identity.Test.Common/Core/Mocks/MockHttpManagerExtensions.cs [hunk 1]
```diff
@@ -350,12 +350,17 @@ public static MockHttpMessageHandler AddAllMocks(this MockHttpManager httpManage
...
             if (userAssignedIdentityId == UserAssignedIdentityId.ClientId)
             {
-                httpMessageHandler.ExpectedQueryParams.Add("client_id", userAssignedClientIdOrResourceId);
+                httpMessageHandler.ExpectedQueryParams.Add("client_id", userAssignedId);
             }
 
             if (userAssignedIdentityId == UserAssignedIdentityId.ResourceId)
             {
-                httpMessageHandler.ExpectedQueryParams.Add("mi_res_id", userAssignedClientIdOrResourceId);
+                httpMessageHandler.ExpectedQueryParams.Add("mi_res_id", userAssignedId);
```
Comment (pmaytak) [inline tests/Microsoft.Identity.Test.Common/Core/Mocks/MockHttpManagerExtensions.cs:358]: Use constants for param name.

File: tests/Microsoft.Identity.Test.Unit/AppConfigTests/ManagedIdentityApplicationBuilderTests.cs [hunk 2]
```diff
@@ -80,27 +81,29 @@ public void TestConstructor_WithCreateUserAssignedId()
...
-                    ManagedIdentityId.WithUserAssignedClientId(userAssignedId) : 
-                    ManagedIdentityId.WithUserAssignedResourceId(userAssignedId))
-                .BuildConcrete();
+            ManagedIdentityApplication mi = CreateMIABuilder(userAssignedId, userAssignedIdentityId).BuildConcrete();
 
             Assert.AreEqual(userAssignedId, mi.ServiceBundle.Config.ClientId);
             Assert.IsNotNull(mi.ServiceBundle.Config.ManagedIdentityId);
 
-            if (isClientId)
+            if (userAssignedIdentityId == UserAssignedIdentityId.ClientId)
```
Comment (pmaytak) [inline tests/Microsoft.Identity.Test.Unit/AppConfigTests/ManagedIdentityApplicationBuilderTests.cs:95]: This `if` can just be one:
`Assert.AreEqual(userAssignedIdentityId, mi.ServiceBundle.Config.ManagedIdentityId.IdType);`

**Review by pmaytak — APPROVED** [review]

File: tests/Microsoft.Identity.Test.Unit/AppConfigTests/ManagedIdentityApplicationBuilderTests.cs (hunk 2 above)
Comment (gladjohn) [reply tests/Microsoft.Identity.Test.Unit/AppConfigTests/ManagedIdentityApplicationBuilderTests.cs]: updated

File: tests/Microsoft.Identity.Test.Common/Core/Mocks/MockHttpManagerExtensions.cs (hunk 1 above)
Comment (gladjohn) [reply tests/Microsoft.Identity.Test.Common/Core/Mocks/MockHttpManagerExtensions.cs]: fixed

**Review by bgavrilMS — APPROVED** [review]

File: tests/Microsoft.Identity.Test.Common/TestConstants.cs [hunk 3]
```diff
@@ -238,6 +238,10 @@ public static string Defaultx5cValue
         public const string MsalArlingtonCCAKeyVaultUri = "https://msidlabs.vault.azure.net:443/secrets/ARLMSIDLAB1-IDLASBS-App-CC-Secret";
         public const string MsalArlingtonCCAKeyVaultSecretName = "ARLMSIDLAB1-IDLASBS-App-CC-Secret";
 
+        public const string ManagedIdentityClientId = "client_id";
+        public const string ManagedIdentityObjectId = "object_id";
+        public const string ManagedIdentityResourceId = "mi_res_id";
```
Comment (pmaytak) [inline tests/Microsoft.Identity.Test.Common/TestConstants.cs:243]: I'd use the ones defined in the `Constants` class.

**Review by pmaytak — APPROVED** [review]

File: tests/Microsoft.Identity.Test.Common/TestConstants.cs (hunk 3 above)
Comment (gladjohn) [reply tests/Microsoft.Identity.Test.Common/TestConstants.cs:243]: let me take a look at it.

**Review by neha-bhargava — DISMISSED** [review]

**Comment by gladjohn** [comment]
Doc updates : https://github.com/MicrosoftDocs/microsoft-authentication-library-dotnet/pull/142

@localden FYI

**Review by neha-bhargava — APPROVED** [review]

**Pull request merged**